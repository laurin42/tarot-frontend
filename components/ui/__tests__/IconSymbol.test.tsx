import React from "react";
import { render } from "@testing-library/react-native";
import { IconSymbol } from "../IconSymbol"; // Adjust path if needed

describe("IconSymbol Component", () => {
  const testSymbolName = "star.fill"; // Use a valid symbol name string directly
  const defaultColor = "black"; // Define a default color for required prop

  it("renders correctly with default props and matches snapshot", () => {
    const tree = render(
      <IconSymbol name={testSymbolName} color={defaultColor} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with custom color and matches snapshot", () => {
    const tree = render(
      <IconSymbol name={testSymbolName} color="blue" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with custom size and matches snapshot", () => {
    const tree = render(
      <IconSymbol name={testSymbolName} size={40} color={defaultColor} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with custom color and size and matches snapshot", () => {
    const tree = render(
      <IconSymbol name={testSymbolName} color="red" size={30} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with weight prop and matches snapshot", () => {
    const tree = render(
      <IconSymbol name={testSymbolName} weight="bold" color={defaultColor} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with all valid props and matches snapshot", () => {
    const tree = render(
      <IconSymbol
        name={testSymbolName}
        size={25}
        weight="semibold"
        color="purple"
        style={{ margin: 5 }} // Add a style test
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
