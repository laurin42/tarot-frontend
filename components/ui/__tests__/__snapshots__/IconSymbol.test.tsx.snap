// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconSymbol Component renders correctly with all valid props and matches snapshot 1`] = `
<ViewManagerAdapter_SymbolModule
  animated={false}
  colors={[]}
  name="star.fill"
  resizeMode="scaleAspectFit"
  style={
    [
      {
        "height": 24,
        "width": 24,
      },
      [
        {
          "height": 25,
          "width": 25,
        },
        {
          "margin": 5,
        },
      ],
    ]
  }
  tintColor="purple"
  type="monochrome"
  weight="semibold"
/>
`;

exports[`IconSymbol Component renders correctly with custom color and matches snapshot 1`] = `
<ViewManagerAdapter_SymbolModule
  animated={false}
  colors={[]}
  name="star.fill"
  resizeMode="scaleAspectFit"
  style={
    [
      {
        "height": 24,
        "width": 24,
      },
      [
        {
          "height": 24,
          "width": 24,
        },
        undefined,
      ],
    ]
  }
  tintColor="blue"
  type="monochrome"
  weight="regular"
/>
`;

exports[`IconSymbol Component renders correctly with custom color and size and matches snapshot 1`] = `
<ViewManagerAdapter_SymbolModule
  animated={false}
  colors={[]}
  name="star.fill"
  resizeMode="scaleAspectFit"
  style={
    [
      {
        "height": 24,
        "width": 24,
      },
      [
        {
          "height": 30,
          "width": 30,
        },
        undefined,
      ],
    ]
  }
  tintColor="red"
  type="monochrome"
  weight="regular"
/>
`;

exports[`IconSymbol Component renders correctly with custom size and matches snapshot 1`] = `
<ViewManagerAdapter_SymbolModule
  animated={false}
  colors={[]}
  name="star.fill"
  resizeMode="scaleAspectFit"
  style={
    [
      {
        "height": 24,
        "width": 24,
      },
      [
        {
          "height": 40,
          "width": 40,
        },
        undefined,
      ],
    ]
  }
  tintColor="black"
  type="monochrome"
  weight="regular"
/>
`;

exports[`IconSymbol Component renders correctly with default props and matches snapshot 1`] = `
<ViewManagerAdapter_SymbolModule
  animated={false}
  colors={[]}
  name="star.fill"
  resizeMode="scaleAspectFit"
  style={
    [
      {
        "height": 24,
        "width": 24,
      },
      [
        {
          "height": 24,
          "width": 24,
        },
        undefined,
      ],
    ]
  }
  tintColor="black"
  type="monochrome"
  weight="regular"
/>
`;

exports[`IconSymbol Component renders correctly with weight prop and matches snapshot 1`] = `
<ViewManagerAdapter_SymbolModule
  animated={false}
  colors={[]}
  name="star.fill"
  resizeMode="scaleAspectFit"
  style={
    [
      {
        "height": 24,
        "width": 24,
      },
      [
        {
          "height": 24,
          "width": 24,
        },
        undefined,
      ],
    ]
  }
  tintColor="black"
  type="monochrome"
  weight="bold"
/>
`;
